dask/dataframe/tests/test_indexing.py

```diff
@@ -34,13 +34,17 @@ def test_loc():
     assert_eq(d.loc[3:], full.loc[3:])
     assert_eq(d.loc[[5]], full.loc[[5]])
     assert_eq(d.loc[[3, 4, 1, 8]], full.loc[[3, 4, 1, 8]])
+    assert_eq(d.loc[[3, 4, 1, 9]], full.loc[[3, 4, 1, 9]])
+    assert_eq(d.loc[np.array([3, 4, 1, 9])], full.loc[np.array([3, 4, 1, 9])])
```

Comment:
@mrocklin I've improved the `isinstance` checks, but I'm not sure how to fix the `.loc[[]]` issue. The culprit is [this line](https://github.com/dask/dask/blob/master/dask/dataframe/indexing.py#L107), and probably an easy fix for someone who knows their way around the code base.